### PR TITLE
SAK-44901: SAMIGO - Importing ZIP with a xml file

### PR DESCRIPTION
**Pull request by jesusmmp**

### CODE REVIEW

File: samigo/samigo-cp/src/java/org/sakaiproject/tool/assessment/contentpackaging/ImportService.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ public String unzipImportFile(String filename) {
 	    	File dir = new File(unzipLocation.toString()); // directory where file would be saved
 	    	if (!dir.exists()) {
 	    		if (!dir.mkdirs()) {
-	    			log.error("unable to mkdir " + dir.getPath());
+	    			log.error("Unable to mkdir " + dir.getPath());
```
Comment (adrianfish) [inline samigo/samigo-cp/src/java/org/sakaiproject/tool/assessment/contentpackaging/ImportService.java:84]: log.error("Unable to mkdir {}", dir.getPath());

File: samigo/samigo-cp/src/java/org/sakaiproject/tool/assessment/contentpackaging/ImportService.java [hunk 2]
```diff
@@ -90,27 +90,27 @@ public String unzipImportFile(String filename) {
...
-	    			String dirName = zipName.substring(0, ix);
+	    			String dirName = entryName.substring(0, ix);
 	    			if (!dirsMade.contains(dirName)) {
 	    				File d = new File(dir.getPath() + "/" + dirName);
 	    				// If it already exists as a dir, don't do anything
 	    				if (!(d.exists() && d.isDirectory())) {
 	    					// Try to create the directory, warn if it fails
 	    					if (!d.mkdirs()) {
-	    						log.error("unable to mkdir " + dir.getPath() + "/" + dirName);
+	    						log.error("Unable to mkdir " + dir.getPath() + "/" + dirName);
```
Comment (adrianfish) [inline samigo/samigo-cp/src/java/org/sakaiproject/tool/assessment/contentpackaging/ImportService.java:106]: I'd move dir.getPath() + "/" + dirName into a variable called directory. It's used up above and you'll then just be able to write:
log.error("Unable to mkdir {}", directory);

File: samigo/samigo-cp/src/java/org/sakaiproject/tool/assessment/contentpackaging/ImportService.java [hunk 3]
```diff
@@ -133,26 +133,16 @@ public String unzipImportFile(String filename) {
 	    				qtiFilename = namedNodeMap.getNamedItem("href").getNodeValue();
 	    			}
 	    			catch (Exception e) {
-	    				log.error("error parsing imsmanifest.xml");
+	    				log.error("Error parsing imsmanifest.xml: " + e.getMessage());
```
Comment (adrianfish) [inline samigo/samigo-cp/src/java/org/sakaiproject/tool/assessment/contentpackaging/ImportService.java:136]: log.error("Error parsing imsmanifest.xml: {}", e.getMessage(), e);

File: samigo/samigo-cp/src/java/org/sakaiproject/tool/assessment/contentpackaging/ImportService.java [hunk 4]
```diff
@@ -183,11 +173,11 @@ else if (entry.getName() != null && entry.getName().trim().endsWith(".xml")) {
 	    return unzipLocation.toString();
 	}
 
-	public String setQTIFilename() {
+	public String getQtiFilename() {
```
Comment (adrianfish) [inline samigo/samigo-cp/src/java/org/sakaiproject/tool/assessment/contentpackaging/ImportService.java:176]: Why bother doing this? You may as well just use Lombok.

**Review by bjones86 — CHANGES_REQUESTED** [review]
Changes requested, as per Adrian's review.

**Review by ern — APPROVED** [review]

**Review by ern — APPROVED** [review]

**Pull request merged**